### PR TITLE
Client Encryption: Adds code to fix breaking changes introduced in public surface client encryption policy APIs

### PR DESCRIPTION
**Pull request by kr-santosh**
## Description

This PR introduces additional client encryption policy constructors to avoid breaking changes required for minor version release.

## Type of change

- [] Bug fix (non-breaking change which fixes an issue)
- [] This change requires a documentation update

### CODE REVIEW

**Review by abhijitpai — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs [hunk 1]
```diff
@@ -51,10 +51,24 @@ public sealed class ClientEncryptionPolicy
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="ClientEncryptionPolicy"/> class.
+        /// The <see cref="PolicyFormatVersion"/> will be set to 1.
+        /// Note: If you need to include partition key or id field paths as part of <see cref="ClientEncryptionPolicy"/>, please set <see cref="PolicyFormatVersion"/> to 2.
+        /// </summary>
+        /// <param name="includedPaths">List of paths to include in the policy definition.</param>        
+        public ClientEncryptionPolicy(IEnumerable<ClientEncryptionIncludedPath> includedPaths)
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs:58]:
```suggestion
        public ClientEncryptionPolicy(IEnumerable<ClientEncryptionIncludedPath> includedPaths): this(includedPaths, 1)
```

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs [hunk 2]
```diff
@@ -51,10 +51,24 @@ public sealed class ClientEncryptionPolicy
...
+        {
+            this.PolicyFormatVersion = 1;
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs:60]:
```suggestion
```

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs [hunk 3]
```diff
@@ -51,10 +51,24 @@ public sealed class ClientEncryptionPolicy
...
+            ClientEncryptionPolicy.ValidateIncludedPaths(includedPaths, this.PolicyFormatVersion);
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs:61]:
```suggestion
```

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs [hunk 4]
```diff
@@ -51,10 +51,24 @@ public sealed class ClientEncryptionPolicy
...
+            this.IncludedPaths = includedPaths;
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs:62]:
```suggestion
```

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - except for the NIT on the ctor overload - I have made suggestions that you can directly merge if you agree.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs (hunk 1 above)
Comment (kr-santosh) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/ClientEncryptionPolicy.cs:58]: Fixed.

**Review by abhijitpai — APPROVED** [review]

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**